src/memory.cpp

```diff
@@ -262,10 +265,12 @@ uint8_t real_emu_read(uint8_t reg)
 		case 5: return gif_recorder_get_state();
 		case 6: return wav_recorder_get_state();
 		case 7: return Options.no_keybinds ? 1 : 0;
-		case 8: return (clockticks6502 >> 0) & 0xff;
-		case 9: return (clockticks6502 >> 8) & 0xff;
-		case 10: return (clockticks6502 >> 16) & 0xff;
-		case 11: return (clockticks6502 >> 24) & 0xff;
+		case 8: 
+		    clock_snap = clockticks6502 - clock_base;
```

Comment:
Windows build needs an explicit cast of `clockticks6502` to `uint32_t`

Reply:
why is clock_snap not defined as uint32_t ? wouldn't that remove the need for a cast?

Reply:
Because we only care about the least significant 32 bits on the 6502 side.  There's no need for all 64 bits.

Reply:
box16 seems to have changed clockticks6502 from 32 to 64 bits though

Reply:
We could make it a 64 bit value and simply ignore that they exist when returning.  That's probably fine too, we're not talking about many host CPU cycles here, and I'm not even sure which one is faster on a 64 bit machine.

Reply:
> box16 seems to have changed clockticks6502 from 32 to 64 bits though

True, but we only care about 32 of those bits here because there are only four 8-bit registers to read the value.   We only care about the lower 32 bits even if the native value is 64 bits.

The spot which you throw away those high bits doesn't really matter though.